impeller/aiks/color_source.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include "flutter/fml/macros.h"
+#include "impeller/entity/contents/runtime_effect_contents.h"
+#include "impeller/entity/entity.h"
+#include "impeller/geometry/color.h"
+#include "impeller/geometry/matrix.h"
+#include "impeller/geometry/point.h"
+#include "impeller/runtime_stage/runtime_stage.h"
+#include "impeller/scene/node.h"
+
+namespace impeller {
+
+struct Paint;
+
+class ColorSource {
+ public:
+  enum class Type {
+    kColor,
+    kImage,
+    kLinearGradient,
+    kRadialGradient,
+    kConicalGradient,
+    kSweepGradient,
+    kRuntimeEffect,
+    kScene,
```

Comment:
Not quite following where the Scene stuff needs to be #ifdef'd out vs. in. Why is this in the clear here, but Scene code under `IMPELLER_ENABLE_3D` in others?

Reply:
Currently we always build Scene as part of Impeller, but completely eliminate it from the DisplayList + UI unless `IMPELLER_ENABLE_3D` is on to make accidental exposure impossible. Maybe we should rename it to e.g. `FLUTTER_ENABLE_3D` some time. 😄